processing/src/main/java/org/apache/druid/query/rowsandcols/ConcatRowsAndColumns.java

```diff
@@ -92,6 +92,9 @@ public Column findColumn(String name)
     if (columnCache.containsKey(name)) {
       return columnCache.get(name);
     } else {
+      if (racBuffer.isEmpty()) {
```

Comment:
while we are at this; can you also add a `Preconditions` check to the constructor?

please also check&test other possibly affected methods as well ; I guess `getColumnNames()` would throw the same error

Reply:
@kgyrtkirk Done